### PR TITLE
Err hard when Dask has mismatched versions or lz4 installed

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes https://github.com/dask/distributed/issues/3932

### CODE REVIEW

File: distributed/tests/test_versions.py [hunk 1]
```diff
@@ -142,3 +144,31 @@ def test_python_version():
...
     assert required["python"] == ".".join(map(str, sys.version_info))
+
+
+@gen_cluster()
+async def test_python_version_error(s, a, b):
+    s.workers[a.address].versions["packages"]["python"] = "3.5.1"
+
+    with pytest.raises(ImportError) as info:
+        async with Client(s.address, asynchronous=True) as client:
+            pass
```
Comment (mrocklin) [inline distributed/tests/test_versions.py:155]: For some reason this `pytest.raises` fails to capture the error on my machine. I'm not sure why.

Comment (abduhbm) [reply distributed/tests/test_versions.py:155]: It fails on my machine too. 

How about switching to `error_message` approach?
```python
def test_python_version_error(kwargs_matching):
    kwargs_matching["workers"]["worker-0"]["packages"]["python"] = "3.5.1"
    msg = error_message(**kwargs_matching)
    assert "Python" in msg["error"]
    assert "major" in msg["error"]
```

**Comment by mrocklin** [comment]
Yes, we could, but I would feel more comfortable testing the user behavior
rather than the internal function if we can find a way to make this work.

**Comment by abduhbm** [comment]
@mrocklin This works too

```python
def test_python_version_error():
    with pytest.raises(ImportError) as info:
        with LocalCluster(
            1,
            processes=False,
            scheduler_port=8786,
            silence_logs=False,
            dashboard_address=None,
        ) as c:
            c.scheduler.workers[c.workers[0].address].versions["packages"][
                "python"
            ] = "3.5.1"
            with Client(c.scheduler.address) as client:
                pass

    assert "Python" in str(info.value)
    assert "major" in str(info.value).lower()
```

**Pull request merged**